src/components/widgets/toolhead/ExtruderMoves.vue

```diff
@@ -123,16 +115,22 @@ export default class ExtruderMoves extends Mixins(StateMixin, ToolheadMixin) {
 
   sendRetractGcode (amount: number, rate: number, wait?: string) {
     if (this.valid) {
+      const safeAmount = Math.min(amount, this.maxExtrudeLength)
+      const safeRate = Math.min(rate, this.maxExtrudeSpeed)
```

Comment:
This logic isn't strictly necessary as `this.valid` does a pretty good job of ensuring correctness before issuing an extrude/retract command. However, I so like this as a guardrail since it allows us to be confident that an extruder command won't exceed machine limits even if there was a logic bug in this form.

Now that I'm typing it out, I'm wondering if Klipper would even respect a command that would exceed its specified limits. I should test that.

At any rate, if Klipper handles this or if it's generally unwanted, I'll remove this portion. Just let me know!

Reply:
I can confirm that Klipper will indeed complain if the value is above the `max_extrude_only_distance` 

![image](https://user-images.githubusercontent.com/85504/218692064-d9d077a6-b52c-400b-94a0-06a3d8e72f96.png)

Reply:
That should not be necessary (to be honest, even the `valid` check is probably overkill here), so I've gone ahead and reverted that small bit before doing the actual merge